dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java

```diff
@@ -486,6 +495,65 @@ public void lifeCycleStarted(LifeCycle event) {
                 }
             }
         });
+
+        environment.lifecycle().addLifeCycleListener(new LifeCycle.Listener() {
+            @Override
+            public void lifeCycleStarted(LifeCycle event) {
+                Session session = hibernate.getSessionFactory().openSession();
+                ManagedSessionContext.bind(session);
+                final Transaction transaction = session.beginTransaction();
+                try {
+                    final String errorPrefix = "could not update old style github token";
+                    TokenDAO tokenDAO = new TokenDAO(hibernate.getSessionFactory());
+                    // cannot use normal github library
+                    final java.net.http.HttpClient client = java.net.http.HttpClient.newBuilder().build();
+                    // exclude both gho tokens that the application generates and ghp tokens that we can insert for testing
+                    // https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/
+                    for (Token t : tokenDAO.findAllGitHubTokens()) {
+                        if (!t.getContent().startsWith("gho_") && !t.getContent().startsWith("ghp_")) {
+                            try {
+                                HttpRequest request = HttpRequest.newBuilder()
+                                    .uri(new URI("https://api.github.com/applications/" + configuration.getGithubClientID() + "/token"))
+                                    .header("Accept", "application/vnd.github+json")
+                                    .header("Authorization", getBasicAuthenticationHeader(configuration.getGithubClientID(), configuration.getGithubClientSecret()))
+                                    .method("PATCH", BodyPublishers.ofString("{\"access_token\":\"" + t.getContent() + "\"}"))
+                                    .build();
+                                final HttpResponse<ResetTokenModel> send = client.send(request, new JsonBodyHandler<>(ResetTokenModel.class));
+                                final ResetTokenModel body = send.body();
+                                if (send.statusCode() == HttpStatus.SC_OK) {
+                                    String newToken = body.token;
+                                    t.setContent(newToken);
+                                    tokenDAO.update(t);
+                                    LOG.info("updated token for {}", t.getUsername());
+                                } else {
+                                    LOG.error(errorPrefix + " for {}, error code {}, token was not found on github", t.getUsername(), send.statusCode());
+                                }
+                            } catch (IOException e) {
+                                LOG.error(errorPrefix + " for {}", t.getUsername());
+                                LOG.error(errorPrefix, e);
```

Comment:
Had the same thought, but unfortunately, there's no such method signature, I think because it could lead to ambiguous calls

i.e.
We have to choose between 
`    public void error(String msg, Throwable t);`
`    public void error(String format, Object... arguments);`

There's no
`    public void error(String format, Object... arguments, Throwable t);`

Reply:
Going to leave this in since the whole section will probably get wiped in 1.13.1 anyway